edx_notifications/channels/urban_airship.py

```diff
@@ -0,0 +1,149 @@
+"""
+NotificationChannelProvider to integrate with the Urban Airship mobile push
+notification services
+"""
+import json
+import logging
+
+import requests
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import RequestException
+
+from edx_notifications.channels.channel import BaseNotificationChannelProvider
+
+# system defined constants that only we should know about
+from settings import MCKA_APP_URL
+
+UA_API_PUSH_ENDPOINT = 'https://go.urbanairship.com/api/push/'
+PUSH_REQUEST_HEADER = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/vnd.urbanairship+json; version=3;'
+}
+
+log = logging.getLogger(__name__)
+
+
+class UrbanAirshipNotificationChannelProvider(BaseNotificationChannelProvider):
+    """
+    Implementation of the BaseNotificationChannelProvider abstract interface
+    """
+
+    def __init__(self, name=None, display_name=None, display_description=None,
+                 link_resolvers=None, application_id=None, rest_api_key=None):
+        """
+        Initializer
+        """
+
+        if not application_id or not rest_api_key:
+            raise Exception('Missing application_id or rest_api_key configuration!')
+
+        self.application_id = application_id
+        self.rest_api_key = rest_api_key
+
+        super(UrbanAirshipNotificationChannelProvider, self).__init__(
+            name=name,
+            display_name=display_name,
+            display_description=display_description,
+            link_resolvers=link_resolvers
+        )
+
+    def dispatch_notification_to_user(self, user_id, msg, channel_context=None):
+        """
+        Send a notification to a user. It is assumed that
+        'user_id' and 'msg' are valid and have already passed
+        all necessary validations
+        :param user_id:
+        :param msg:
+        :param channel_context:
+        :return:
+        """
+
+        obj = {
+            'notification': {'alert': msg.payload['excerpt']},
+            'audience': {'named_user': user_id},
+            'device_types': ['ios', 'android']
```

Comment:
You will also need to explicitly set the device_types because Named Users are not supported on every platform
Ref: https://support.urbanairship.com/hc/en-us/community/posts/208269303-Target-Named-User